backend/src/utils/security/jwt_token.py

```diff
@@ -0,0 +1,140 @@
+from datetime import datetime, timedelta
+from functools import cached_property
+from typing import Any
+
+import jwt
+
+from src.core.config.settings import settings
+from src.core.enums.jwt_token import TokenEnum
+from src.models import User
+
+
+class JWTToken:
+    """
+    JWTToken is a utility class for handling JSON Web Tokens (JWT) for authentication and authorization purposes.
+
+    Attributes:
+        private_key (str): The private key used for signing the JWT tokens.
+        public_key (str): The public key used for verifying the JWT tokens.
+        algorithm (str): The algorithm used for encoding and decoding the JWT tokens.
+        issuer (str): The issuer of the JWT tokens, typically identifying the application or service issuing the tokens.
+        expire_minutes (timedelta): The expiration time for access tokens.
+        expire_days (timedelta): The expiration time for refresh tokens.
+    """
+
+    private_key: str = settings.AUTH_SETTINGS.PRIVATE_KEY_PATH.read_text()
+    public_key: str = settings.AUTH_SETTINGS.PUBLIC_KEY_PATH.read_text()
+    algorithm: str = settings.AUTH_SETTINGS.ALGORITHM
+    issuer: str = settings.AUTH_SETTINGS.TOKEN_ISSUER
+    expire_minutes: timedelta = settings.AUTH_SETTINGS.ACCESS_TOKEN_EXPIRE_MINUTES
+    expire_days: timedelta = settings.AUTH_SETTINGS.REFRESH_TOKEN_EXPIRE_DAYS
+
+    @cached_property
+    def set_jwt_token_header(self) -> dict[str, str]:
+        """
+        Returns the JWT token header.
+
+        Returns:
+            dict[str, str]: The JWT token header with type and algorithm.
+        """
+        return {"typ": "JWT", "alg": self.algorithm}
+
+    def update_payload(self, payload: dict, expire: timedelta) -> dict:
+        """
+        Updates the given payload with expiration (`exp`), not-before (`nbf`),
+        and issued-at (`iat`) times, and adds the issuer (`iss`).
+
+        Args:
+            payload (dict): The original payload.
+            expire (timedelta): The expiration time to be added to the payload.
+
+        Returns:
+            dict: The updated payload with `exp`, `nbf`, `iat`, and `iss` fields.
+        """
+        payload_copy = payload.copy()
+        expire = datetime.now() + expire
+        payload_copy.update(
+            {
+                "iss": self.issuer,
+                "exp": expire,
+                "nbf": datetime.now(),
+                "iat": datetime.now(),
+            }
+        )
+        return payload_copy
```

Comment:
```suggestion
        payload_copy = payload.copy()
        
        return payload_copy.update(
            {
                "iss": self.issuer,
                "exp": datetime.now() + expire,
                "nbf": datetime.now(),
                "iat": datetime.now(),
            }
        )
```

Reply:
I made `datetime_now: datetime = datetime.now()` as a class attribute and used that in this func